objdb/etcdClient.go

```diff
@@ -165,23 +151,19 @@ func (ep *EtcdClient) ListDir(key string) ([]string, error) {
 		}
 	}
 
-	if !resp.Node.Dir {
-		log.Errorf("ListDir response is not a directory")
-		return nil, errors.New("Response is not directory")
-	}
-
 	var retList []string
-	// Call a recursive function to recurse thru each directory and get all files
-	// Warning: assumes directory itep is not interesting to the caller
-	// Warning2: there is also an assumption that keynames are not required
-	//           Which means, caller has to derive the key from value :(
-	retList = recursAddNode(resp.Node, retList)
+
+	// convert all the keys into strings (etcd3 doesn't have directories)
+	for _, kv := range resp.Kvs {
+		retList = append(retList, string(kv.Key))
```

Comment:
It's should be get Value from kv, not Key. Like this:
retList = append(retList, string(kv.Value)).
In the mvccpb, It's define to the struct, 
        `type KeyValue struct {
	// key is the key in bytes. An empty key is not allowed.
	Key []byte `protobuf:"bytes,1,opt,name=key,proto3" json:"key,omitempty"`
	// create_revision is the revision of last creation on this key.
	CreateRevision int64 `protobuf:"varint,2,opt,name=create_revision,json=createRevision,proto3" json:"create_revision,omitempty"`
	// mod_revision is the revision of last modification on this key.
	ModRevision int64 `protobuf:"varint,3,opt,name=mod_revision,json=modRevision,proto3" json:"mod_revision,omitempty"`
	// version is the version of the key. A deletion resets
	// the version to zero and any modification of the key
	// increases its version.
	Version int64 `protobuf:"varint,4,opt,name=version,proto3" json:"version,omitempty"`
	// value is the value held by the key, in bytes.
	Value []byte `protobuf:"bytes,5,opt,name=value,proto3" json:"value,omitempty"`
	// lease is the ID of the lease that attached to key.
	// When the attached lease expires, the key will be deleted.
	// If lease is 0, then no lease is attached to the key.
	Lease int64 `protobuf:"varint,6,opt,name=lease,proto3" json:"lease,omitempty"`
}`